estate/models/estate_property.py

```diff
@@ -0,0 +1,104 @@
+from odoo import api, fields, models
+from odoo.exceptions import UserError, ValidationError
+from odoo.tools.float_utils import float_compare
+
+import datetime
+
+class EstateProperty(models.Model):
+    _name = "estate.property"
+    _description = "Estate property"
+
+    name = fields.Char('Name', required=True)
+    tag_ids = fields.Many2many('estate.property.tag', string='Tags')
+    property_type_id = fields.Many2one('estate.property.type', string='Property Type', required=True)
+    partner_id = fields.Many2one('res.partner', string='Buyer', copy=False)
+    user_id = fields.Many2one('res.users', string='Salesperson', default=lambda self: self.env.user)
+    offer_ids = fields.One2many('estate.property.offer', 'property_id', string='Offers')
+    best_offer = fields.Float('Best Offer', readonly=True, compute='_compute_best_offer')
+    description = fields.Text('Description')
+    postcode = fields.Char('Postcode')
+    date_availability = fields.Date('Availability Date', copy=False, default=fields.Date.add(fields.Date.today(), months=3))
+    expected_price = fields.Float('Expected Price', required=True)
+    selling_price = fields.Float('Selling Price', readonly=True, copy=False)
+    bedrooms = fields.Integer('Bedrooms', default=2)
+    living_area = fields.Integer('Living Area')
+    facades = fields.Integer('Facades')
+    garage = fields.Boolean('Garage')
+    garden = fields.Boolean('Garden')
+    garden_area = fields.Integer('Garden Area')
+    garden_orientation = fields.Selection(
+        string='Garden Orientation',
+        selection=[
+            ('north', 'North'),
+            ('west', 'West'),
+            ('south', 'South'),
+            ('east', 'East'),
+        ]
+    )
+    total_area = fields.Integer('Total Area', compute='_compute_total_area', readonly=True)
+    state = fields.Selection(
+        selection=[
+            ('new', 'New'),
+            ('offer_received', 'Offer Received'),
+            ('offer_accepted', 'Offer Accepted'),
+            ('sold', 'Sold'),
+            ('cancelled', 'Cancelled'),
+        ],
+        default='new',
+    )
+
+    active = fields.Boolean('Active', default=True)
+
+    _sql_constraints = [
+        ('selling_price', 'CHECK(selling_price > 0)', 'Prices must be strictly positive.'),
+        ('expected_price', 'CHECK(expected_price > 0)', 'Prices must be strictly positive.'),
+    ]
+
+    _order = "id desc"
+
+    def action_set_cancelled(self):
+        if self.state == 'sold':
+            raise UserError('Sold properties can not be cancelled')
+        else:
+            self.write({'state': 'cancelled'})
```

Comment:
we usually use the write function when there are multiple fields to set, but as you only have one, the cleanest solution is to directly do `self.state = 'cancelled'`